presto-main/src/test/java/com/facebook/presto/operator/TestHashAggregationOperator.java

```diff
@@ -657,6 +677,64 @@ public void testSpillerFailure()
         }
     }
 
+    @Test
+    public void testMemoryPools()
+            throws Exception
+    {
+        testMemoryPools(Step.PARTIAL, Optional.empty(), false, true);
+        testMemoryPools(Step.PARTIAL, Optional.of(new DataSize(1, KILOBYTE)), false, true);
+        testMemoryPools(Step.SINGLE, Optional.empty(), false, false);
+        testMemoryPools(Step.INTERMEDIATE, Optional.empty(), false, true);
+        testMemoryPools(Step.FINAL, Optional.empty(), false, false);
+        testMemoryPools(Step.SINGLE, Optional.empty(), true, true);
+        testMemoryPools(Step.INTERMEDIATE, Optional.empty(), true, true);
+        testMemoryPools(Step.FINAL, Optional.empty(), true, true);
+    }
+
+    private void testMemoryPools(Step step, Optional<DataSize> maxPartialMemory, boolean useSystemMemoryPoolFlag, boolean expectedToUseSystemMemoryPool)
```

Comment:
ditto regarding the names of the boolean parameters and the confusion I had above.

Reply:
That code was removed